### ISSUE TITLE
[HOLD https://github.com/Expensify/Auth/pull/12960] [$250] Workspace-As admin,opening expense details page in archived control WS directs to hmm not here

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#49088]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.33-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal team

## Action Performed:
1. Launch app
2. Create a workspace and invite a member as employee
3. Tap on a category - gl code
4. Upgrade to control workspace
5. Enter gl and payroll code and save it
6. From settings, select members must categorize all expenses
7. In mweb, login as employee
8. Open employee workspace chat
9. Create a expense with big amount, future date
10. As admin, delete this control workspace
11. Note the expense details page in archived workspace of both employee and admin

## Expected Result:
As admin, opening expense details page in archived control workspace must not direct to hmm not here .

## Actual Result:
As admin, opening expense details page in archived control workspace directs to hmm not here but as employee can view expense.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/267f3c2a-765e-48df-9a4f-7404e69921a1

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021836148946035230114</li>
        <li>Upwork Job ID: 1836148946035230114</li>
        <li>Last Price Increase: 2024-09-24</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reviewer | 104114173</li>
</ul></li>
    </ul>
</details>

**Comment by burczu** [comment]
Hi, I’m Bartek from Callstack and I would like to investigate this issue.

**Comment by burczu** [comment]
Thanks for assigning me to the issue. I was able to reproduce the issue and from what I see for admin, when we try to access the expense details report, we get the `Report not found` error from the backend:

![Screenshot 2024-09-27 at 11 23 01](https://github.com/user-attachments/assets/6bf92e93-2be8-4124-b36e-e679903d958d)

So I suspect this issue may be related to the backend - especially we don't get this error for an employee, so it looks like the report is not accessible in circumstances described in this issue. I'll spend some more time digging deeper to check if there is something else on the frontend side that may cause this, but I didn't find anything promising yet.

**Comment by johncschuster** [comment]
Thanks, @burczu! I'll take it internal then!

**Comment by johncschuster** [comment]
bump

**Comment by johncschuster** [comment]
Thanks, @shubham1206agra! I've done that now.

**Comment by johncschuster** [comment]
Waiting on this to get picked up

**Comment by johncschuster** [comment]
Still waiting on this one to get picked up

**Comment by trjExpensify** [comment]
@srikarparsi curious for your input here, seems like another reason to execute this plan to not archive expense reports on workspace deletion: https://expensify.slack.com/archives/C036QM0SLJK/p1728047968818889?thread_ts=1728047962.737329&cid=C036QM0SLJK

Moving to the #quality GH project where the remaining "closed and archived" work has moved.

**Comment by johncschuster** [comment]
Not overdue, Melvin. We're looking for some insight from @srikarparsi.

**Comment by srikarparsi** [comment]
> @srikarparsi curious for your input here, seems like another reason to execute this plan to not archive expense reports on workspace deletion:

So it's kind of two different things in the code at least.
 - Archiving expense reports and transaction threads happens [here](https://github.com/Expensify/Auth/blob/31fe7a17aec09beaada12344e8ed73c3aee6226d/auth/lib/Policy.cpp#L3480-L3484) This is going to stop once this [PR](https://github.com/Expensify/Auth/pull/12729) gets merged.
 - Unsharing open expense reports from admins and auditors happens [here](https://github.com/Expensify/Auth/blob/main/auth/lib/Policy.cpp#L3490-L3499)
 
 I think this bug is happening because of the unsharing part. Maybe we should stop unsharing open reports from admins?

**Comment by trjExpensify** [comment]
>I think this bug is happening because of the unsharing part.

Intereeesting. Are you saying these open expense reports remain on the workspace, but the admins/auditors of said workspace lose access to them? 🤔

**Comment by johncschuster** [comment]
Not overdue. Still discussing

**Comment by johncschuster** [comment]
@srikarparsi can you take a look at Tom's last message when you get a moment?

**Comment by srikarparsi** [comment]
> Intereeesting. Are you saying these open expense reports remain on the workspace, but the admins/auditors of said workspace lose access to them? 🤔

Yup, exactly. This comment: [Unshare all open reports from admins and auditors.](https://github.com/Expensify/Auth/blob/a995844e100b08b34c24aec4abb0883e43af95ec/auth/lib/Policy.cpp#L3489)

**Comment by trjExpensify** [comment]
Yeah, so I think the question comes down to: 

**Q: When a workspace is deleted, should the admins/auditors retain access to the open expense reports on the workspace to be able to move those somewhere else?** 

1. If yes, then we shouldn't unshare them and allow the admins to `Move report` somewhere else (i.e another workspace that they're an admin of). 
2. If no, then we should continue to unshare them with anyone but the submitter (who can `Move report` elsewhere, including an option for the selfDM which would unreport the expenses).

I don't really see the downside of (1). Do you, @JmillsExpensify @twisterdotcom? 

Another option might have been to automatically move the `open` expense reports off the workspace on deletion and to the member's individual workspace, but the notion of a "personal workspace" in NewDot doesn't exist. The closest equivalent would be the selfDM, but that would mean auto-unreporting all the expenses (as only unreported expenses exist in the selfDM), which I'm not sure a submitter would appreciate more than just keeping the open report intact and being able to move it elsewhere to another workspace or unreport the expenses on their own accord.

**Comment by johncschuster** [comment]
(Still discussing, Melvin)

**Comment by twisterdotcom** [comment]
I do prefer leaving the former admins with some access, and I don't see any problem with doing that.

**Comment by johncschuster** [comment]
Discussing

**Comment by trjExpensify** [comment]
>I do prefer leaving the former admins with some access, and I don't see any problem with doing that.

Cool, then I think we do (1) then. 👍

**Comment by johncschuster** [comment]
Sounds good! @srikarparsi, will you be taking this on?

**Comment by johncschuster** [comment]
I believe @srikarparsi's plate is pretty full at the moment. Should we get this issue redistributed, or can it wait until his bandwidth opens back up?

**Comment by trjExpensify** [comment]
Srikar isn't assigned in the first place, so it's available as a Hot pick in the meantime if someone can pick it up.

**Comment by srikarparsi** [comment]
Hey sorry! I started looking at this and there's a little more to unpack because we also unshare all archived reports from everyone except the owner [here](https://github.com/Expensify/Auth/blob/698a3b76258975488e0489a44cd50e899c72615b/auth/lib/Policy.cpp#L3575-L3577). I'm going to think about this tomorrow and might have a couple of questions about expected behavior but I think I can take this on.

**Comment by srikarparsi** [comment]
Ok, so we do two things in terms of un-sharing:

1. When a policy is deleted, we un-share all open reports with admins and auditors.
2. When a policy is deleted, we un-share all archived reports from everyone except the policy owner.

We're still archiving expense reports in the context of the code (they will still have the private_isArchived rNVP so that we can know to hide action buttons - submit, pay, etc. - in the frontend)

I know we had some other discussions about what reports should be accessible to who after a policy has been deleted. @trjExpensify or @twisterdotcom do you have any thoughts on this and I can move forward with implementing it?

**Comment by trjExpensify** [comment]
I'm not sure what exactly you're asking for thoughts on, but to summarise [on this](https://github.com/Expensify/App/issues/49088#issuecomment-2435405397). 

- When a policy is deleted, don't modify the expense report's state or unshare it with the participants of the report. Leave that alone, just hide the `Submit`, `Approve`, or `Pay` actions (which ever is applicable to the report state) and prevent additional expenses from being added to it. This ensures that:
     - Backwards compatibility with OldDot is maintained for report commenting, because they can still be added to an expense report. 
     - We prevent actions that result in immediate errors because they can't be taken on a report that's associated with a deleted workspace 
     - Users maintain access to the expense reports, so they can move them to another workspace in light of this policy being deleted (pending other restrictions like report state, exported status etc) to progress elsewhere.

**Comment by srikarparsi** [comment]
> I'm not sure what exactly you're asking for thoughts on

Sorry, the part I wanted thoughts on was who loses access to each type of report after a policy has been deleted. So for example right now I believe:

Owners: Lose access to all open expense reports
Admins: Lose access to all archived reports
Auditors: Lose access to all archived reports
Members: Lose access to all archived reports

And based on what you said above, is expected that:
Owners: Retain access to all reports
Admins: Lose access to all archived chat and task reports (not expense reports)
Auditors: Lose access to all archived chat and task reports (not expense reports)
Members: Lose access to all archived chat and task reports (not expense reports)

Is this right? Sorry for not clarifying earlier.

**Comment by trjExpensify** [comment]
Right, we aren't looking to modify the existing behaviour of other report types on policy deletion. We're clarifying what to do about `expense` reports only [with this](https://github.com/Expensify/App/issues/49088#issuecomment-2459735609).

**Comment by srikarparsi** [comment]
Cool started on a PR [here](https://github.com/Expensify/Auth/pull/13082)

**Comment by johncschuster** [comment]
Chill out, Melvin. A PR is in the works!

**Comment by johncschuster** [comment]
bump

**Comment by johncschuster** [comment]
bump

**Comment by srikarparsi** [comment]
We are holding this for https://github.com/Expensify/Auth/pull/12960

**Comment by srikarparsi** [comment]
on hold

**Comment by johncschuster** [comment]
@srikarparsi should we downgrade this to `Weekly` while it's on hold?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by johncschuster** [comment]
Looks like discussion is still ongoing in the above linked PR

**Comment by johncschuster** [comment]
@srikarparsi what's the latest on this one?

**Comment by srikarparsi** [comment]
Hey @johncschuster, still waiting on this [PR](https://github.com/Expensify/App/pull/52183) to get merged which should hopefully be this week.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fourth week)

**Comment by srikarparsi** [comment]
Left a comment asking [here](https://github.com/Expensify/App/pull/52183#issuecomment-2571721801)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fifth week)

**Comment by srikarparsi** [comment]
Yup, we fixed it in this [PR](https://github.com/Expensify/Auth/pull/13695) so we can close this out.
